runtime/src/main/java/io/quarkiverse/openapi/generator/providers/OAuth2AuthenticationProvider.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkiverse.openapi.generator.providers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.oidc.client.filter.OidcClientRequestFilter;
+
+public class OAuth2AuthenticationProvider extends OidcClientRequestFilter implements AuthProvider {
+
+    private final String clientId;
+    private final List<OperationAuthInfo> applyToOperations = new ArrayList<>();
+
+    public OAuth2AuthenticationProvider(final String clientId) {
+        this.clientId = clientId;
+        // it's fine calling it here since this class will be created on `init()` method of the generated CompositeAuthenticationProvider
+        super.init();
+    }
+
+    @Override
+    protected Optional<String> clientId() {
+        return Optional.of(this.clientId);
+    }
+
+    @Override
+    public String getName() {
+        return this.clientId;
+    }
+
+    @Override
+    public List<OperationAuthInfo> operationsToFilter() {
+        return applyToOperations;
+    }
+
+    @Override
+    public AuthProvider addOperation(OperationAuthInfo operationAuthInfo) {
+        this.applyToOperations.add(operationAuthInfo);
```

Comment:
Yeah, I thought about this, but I'm trying to avoid excessive inheritance. I'll do an aggregation instead with a helper class. These filters can't have too much inheritance to add space for users to overwrite/add their custom logic if needed.